python/ray/tune/schedulers/resource_changing_scheduler.py

```diff
@@ -14,13 +17,337 @@
 logger = logging.getLogger(__name__)
 
 
-class _DistributeResources:
-    """Generic functionality for resource allocation functions"""
+@PublicAPI(stability="beta")
+class DistributeResources:
```

Comment:
How does this play in the context of autoscaling, which is a totally different mechanism.

Reply:
When a cluster scales up or down, the resource allocation function will take that into account, as it checks the available resources every time it is called